File: pageobjects/src/test/java/autoTestJavaFullObjects/CommentTest.java
package autoTestJavaFullObjects;

import autoTestJavaFullObjects.pageObjects.WpPageMain;
import autoTestJavaFullObjects.pageObjects.WpPagePost;
import org.junit.Assert;
import org.junit.Test;

import java.util.UUID;

/**
 * Created by Rafal on 2017-05-31.
 */
public class CommentTest extends BaseTests {

    @Test
    public void ShouldAddCommentToThirdPost() throws InterruptedException {

        // GIVEN / ARRANGE
        String comment = generateRandomText();
        String email = UUID.randomUUID().toString() + "@test.com";
        String name = UUID.randomUUID().toString();

        //WHEN / ACT
        WpPageMain mainPage = new WpPageMain(this.driver);
        mainPage.Open();
        WpPagePost postPage = mainPage.DisplayPost(3);
        postPage.AddComment(comment, email, name);

        //THEN / ASSERT
        Assert.assertTrue("Comment should be posted", postPage.AssertCommentIsPosted(comment, name));
    }

}

File: C#PageObjects/VisualStudioPageObjects/autoTestJavaFullObjects/CommentTest.cs
using autoTestJavaFullObjects.PageObjects;
using Xunit;

namespace autoTestJavaFullObjects
{
    public class CommentTest : BaseTests
    {
        [Fact]
        public void ShouldAddCommentToThirdPost()
        {
            // GIVEN / ARRANGE
            string commentText = GenerateRandomText();
            string commentAuthorEmail = GenerateRandomEmail();
            string commentAuthorName = GenerateRandomName();

            //WHEN / ACT
            WpMainPage mainPage = new WpMainPage(this.driver);
            mainPage.Open();
            WpPostPage postPage = mainPage.DisplayPost(3);
            postPage.AddComment(commentText, commentAuthorEmail, commentAuthorName);

            //THEN / ASSERT
            Assert.True(postPage.IsCommentPosted(commentText, commentAuthorName));
        }

        [Fact]
        public void ShouldAddReplyToComment()
        {
            // GIVEN / ARRANGE
            string commentText = GenerateRandomText();
            string commentAuthorEmail = GenerateRandomEmail();
            string commentAuthorName = GenerateRandomName();

            string replyText = GenerateRandomText();
            string replyAuthorEmail = GenerateRandomEmail();
            string replyAuthorName = GenerateRandomName();

            //WHEN / ACT
            WpMainPage mainPage = new WpMainPage(this.driver);
            mainPage.Open();
            WpPostPage postPage = mainPage.DisplayPost(1);
            postPage.AddComment(commentText, commentAuthorEmail, commentAuthorName);
            postPage.AddReplyToComment(commentText, commentAuthorName, replyText, replyAuthorEmail, replyAuthorName);

            //THEN / ASSERT
            Assert.True(postPage.IsReplyPosted(commentText, commentAuthorName, replyText, replyAuthorName));
        }
    }
}
